src/adaptors/in-memory-articles.ts

```diff
@@ -1,15 +1,24 @@
 import { Iri, JsonLdObj } from 'jsonld/jsonld-spec';
+import { schema } from 'rdf-namespaces';
 import Articles from '../articles';
 import ArticleHasNoId from '../errors/article-has-no-id';
 import ArticleNotFound from '../errors/article-not-found';
+import ArticleHasNoTitle from '../errors/article-has-no-title';
+import ArticleHasIncorrectType from '../errors/article-has-incorrect-type';
 
 export default class InMemoryArticles implements Articles {
   private articles: { [key: string]: JsonLdObj } = {};
 
   async add(article: JsonLdObj): Promise<void> {
+    if (!('@type' in article) || !(article['@type'].includes(schema.Article))) {
+      throw new ArticleHasIncorrectType(('@type' in article) ? article['@type'] : undefined);
+    }
     if (!('@id' in article)) {
       throw new ArticleHasNoId();
     }
+    if (!(schema.name in article) || article[schema.name].length === 0) {
+      throw new ArticleHasNoTitle();
+    }
```

Comment:
While this shouldn't expand to other validation steps, suggest we keep this one at the HTTP layer (the other two are required for persistence, this one isn't).